### PR TITLE
[NEW] WebRTC Call session

### PR DESCRIPTION
**Pull request by djcruz93**
## Proposed changes (including videos or screenshots)

Create a webRTC call session and save it in the form of a message. It involves the creation of a `webrtc:createSession` method that is called by the client-side on initiating the call. The flow is on the same lines as that of Jitsi.

### CODE REVIEW

File: app/webrtc/server/methods/webrtcCreateSession.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import { Meteor } from 'meteor/meteor';
+import { TAPi18n } from 'meteor/rocketchat:tap-i18n';
+
+import { Messages, Users } from '../../../models/server';
+import { callbacks } from '../../../callbacks/server';
+import { canSendMessage } from '../../../authorization/server';
+import { SystemLogger } from '../../../logger/server';
+
+Meteor.methods({
```
Comment (murtaza98) [inline app/webrtc/server/methods/webrtcCreateSession.js:9]: IMO we should avoid using meteor methods and build these type of functionality using endpoints

Comment (murtaza98) [reply app/webrtc/server/methods/webrtcCreateSession.js:9]: In noticed we already have an endpoint defined [here](https://github.com/RocketChat/Rocket.Chat/blob/2fa2ba0fdd6cd29fed29c1b1118ce89e54d5856e/app/livechat/server/api/v1/videoCall.js#L10) for starting a call so IMO this is a good time to modify that endpoint as well to support webrtc calls and use the same on the UI

File: app/webrtc/client/tabBar.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+	return useMemo(() => (enabled ? {
+		groups: ['live'],
+		id: 'webRTCVideo',
+		title: 'WebRTC_Call',
+		icon: 'phone',
+		action: handleClick,
+		full: true,
+		order: 4,
+	} : null), [enabled]);
```
Comment (murtaza98) [inline app/webrtc/client/tabBar.tsx:24]: @djcruz93 Did you find a solution for this warning?
image

File: app/webrtc/server/methods/webrtcCreateSession.js (hunk 1 above)
Comment (murtaza98) [reply app/webrtc/server/methods/webrtcCreateSession.js:9]: @djcruz93 Any updates on this part?

File: app/webrtc/client/tabBar.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import { useMemo } from 'react';
```
Comment (murtaza98) [inline app/webrtc/client/tabBar.tsx:1]: Please fix the conflicts in this file

**Comment by djcruz93** [comment]
@murtaza98 Could you please review this PR? I have pushed the changes as requested.

**Comment by djcruz93** [comment]
Closed in favor of #22689

**Comment by djcruz93** [comment]
Closed in favor of #22689

**Pull request closed without merging**